drivers/ethernet/eth_esp32.c

```diff
@@ -17,6 +17,9 @@
 #include <esp_mac.h>
 #include <hal/emac_hal.h>
 #include <hal/emac_ll.h>
+#if DT_INST_NODE_HAS_PROP(0, ref_clk_output_gpios)
+#include <soc/rtc.h>
+#endif
```

Comment:
Do we need to keep `#ifdef` here? Build testing would probably benefit from unconditional include. And code would be cleaner. The only downside is more preprocessing time, but that is usually not an issue with Zephyr.

Reply:
We don't, I just wanted not to include this header if there is no need.

Reply:
There would be many `#ifdef` statements across whole Zephyr codebase if that would be the default policy. I think that increasing readability by the cost of performance is a good choice.